### PR TITLE
Ensure request is chained before payload is logged

### PR DESCRIPTION
**Pull request by adriangonz**
**What this PR does / why we need it**:
Ensure responsed is chained to the next request before logging the request. Otherwise, we will log as "request" the response from the previous model in the inference graph. In the V2 payload, since the request and response have different fields, this currently ends up in an error later on at the request logger. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **adriangonz**
You can assign the PR to them by writing `/assign @adriangonz` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: executor/licenses/dep.txt
Comment (axsaucedo) [inline executor/licenses/dep.txt:231]: Is this from a missed license run from a different PR?

File: executor/predictor/predictor_process.go
Comment (axsaucedo) [inline executor/predictor/predictor_process.go:103]: This makes sense, also for the Tensorflow protocol - thinking it may be worth also implementing this for the `transformOutput` method?

**Comment by seldondev** [comment]
@adriangonz: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 2a5414a5f554664a9c67c3c44889111791565676 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-4301-integration-x879x) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: executor/predictor/predictor_process.go
Comment (SachinVarghese) [reply executor/predictor/predictor_process.go:103]: Here if the chain function fails and throws an error, the message will not be logged, is that a concern?

File: executor/licenses/dep.txt
Comment (adriangonz) [reply executor/licenses/dep.txt:231]: It seems that way. Either that, or some subdep got updated. The linter failed within the GH Action, and re-generating the licenses file locally fixed it.

File: executor/predictor/predictor_process.go
Comment (adriangonz) [reply executor/predictor/predictor_process.go:103]: Good spot @axsaucedo , just added the same for `transformOutput`.

@SachinVarghese that's a good point. Although if the chaining fails, the request will also fail. Should we log the request if we know it's gonna fail?

Comment (SachinVarghese) [reply executor/predictor/predictor_process.go:103]: I had a look at this again, current behavior seems to be good.

**Review by SachinVarghese — APPROVED** [review]
LGTM

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[SachinVarghese](https://github.com/SeldonIO/seldon-core/pull/4301#pullrequestreview-1095068777 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [SachinVarghese]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [4301]: [Method Not Allowed]

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [4301]: [Method Not Allowed]

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [4301]: [Method Not Allowed]

**Comment by seldondev** [comment]
Failed to merge this PR due to:
>failed merging [4301]: [Method Not Allowed]

**Pull request merged**
